src/common/meta/src/ddl/drop_database/cursor.rs

```diff
@@ -66,31 +65,40 @@ impl DropDatabaseCursor {
         ctx: &mut DropDatabaseContext,
         table_name: String,
         table_id: TableId,
-        table_route_value: DeserializedValueWithBytes<TableRouteValue>,
+        table_route_value: TableRouteValue,
     ) -> Result<(Box<dyn State>, Status)> {
-        match (self.target, table_route_value.get_inner_ref()) {
-            (DropTableTarget::Logical, TableRouteValue::Logical(_))
-            | (DropTableTarget::Physical, TableRouteValue::Physical(_)) => {
-                // TODO(weny): Maybe we can drop the table without fetching the `TableInfoValue`
-                let table_info_value = ddl_ctx
+        match (self.target, table_route_value) {
+            (DropTableTarget::Logical, TableRouteValue::Logical(route)) => {
+                let table_id = route.physical_table_id();
+
+                // Safety: it must be the physical table route.
+                let table_route = ddl_ctx
                     .table_metadata_manager
-                    .table_info_manager()
+                    .table_route_manager()
+                    .table_route_storage()
```

Comment:
Better to use: https://github.com/GreptimeTeam/greptimedb/blob/343525dab8a978a7bfc7df567e8ef6b30f2ee563/src/common/meta/src/key/table_route.rs#L290